pkgs/development/tools/misc/segger-jlink/default.nix

```diff
@@ -0,0 +1,128 @@
+{ lib
+, stdenv
+, fetchurl
+, autoPatchelfHook
+, makeWrapper
+, qt4
+, libudev
+, system
+, config
+, acceptLicense ? config.segger-jlink.acceptLicense or false
+}:
+
+let
+  supported = {
+    x86_64-linux = {
+      name = "x86_64";
+      sha256 = "19gh6hlpq4dm3ji9wdf9n435zimybvgkzshdww25j9bdm0dl0595";
+    };
+    i686-linux = {
+      name = "i386";
+      sha256 = "1f0iik519panf3649nxszzh61vss927k7zgr4yx1lj991ba6rjdq";
+    };
+    aarch64-linux = {
+      name = "arm64";
+      sha256 = "0gyka0w7xiphp22vkszkx8bvm46k3pnsxmpg0mzi580gpwh3ay1q";
+    };
+    armv7l-linux = {
+      name = "arm";
+      sha256 = "0ws8l85fl4slvx6d4q779f24ppdwm901kbdbksjl66zn25wz8g33";
+    };
+  };
+
+  platform = supported.${system} or (throw "unsupported platform ${system}");
+
+  throwLicense = throw ''
+    Use of the "SEGGER JLink Software and Documentation pack" requires the
+    acceptance of the following licenses:
+
+      - SEGGER Downloads Terms of Use [1]
+      - SEGGER Software Licensing [2]
+
+    You can express acceptance by setting acceptLicense to true in your
+    configuration. Note that this is not a free license so it requires allowing
+    unfree licenses as well.
+
+    configuration.nix:
+      nixpkgs.config.allowUnfree = true;
+      nixpkgs.config.segger-jlink.acceptLicense = true;
+
+    config.nix:
+      allowUnfree = true;
+      segger-jlink.acceptLicense = true;
+
+    [1]: https://www.segger.com/downloads/jlink/JLink_Linux_V720_x86_64.tgz
+    [2]: https://www.segger.com/purchase/licensing/
+  '';
+
+in stdenv.mkDerivation {
+  pname = "segger-jlink";
+  version = "7.20";
```

Comment:
You could use  720 as the version name, you could then use a variable in the url.

Reply:
I was a bit hesitant, because the website officially uses that form. But given that it is really the download link that does not specify this better, it would be a cosmetic thing at best, so I went with your approach in the end.